.github/workflows/contrib-tests.yml

```diff
@@ -17,8 +17,26 @@ concurrency:
 permissions: read-all
 
 jobs:
+  contrib-tests-prepare:
+    runs-on: ubuntu-latest
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'Skip Contrib Tests') }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Prepare Contrib Tests
+        run: |
+          contrib_path=/tmp/opentelemetry-collector-contrib
+          git clone --depth=1 https://github.com/open-telemetry/opentelemetry-collector-contrib.git $contrib_path
+          make CONTRIB_PATH=$contrib_path prepare-contrib
+      - uses: actions/upload-artifact@v4
```

Comment:
Doesn't it mean that we will be testing against one contrib version for 90 days?

Reply:
I'm looking at https://github.com/actions/upload-artifact?tab=readme-ov-file#retention-period. Should we add git sha in the `contrib_path`?

Reply:
ouch. I'll open a new PR to add that

Reply:
Why do we need to upload it then? Should we use https://github.com/actions/cache instead?

Reply:
I followed the recipe I have seen used most of the time. A cache is a step better I didn't look into because I was afraid of having cache issues. That would shave a few dozen more seconds off the build.